.github/workflows/components-install.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+use_path_repository_in_composer () {
+    cd $1
+
+    sed -i 's/"require": {/"repositories": [{"type": "path", "url": "..\/*"}], \0/' composer.json
+    sed -i 's/\("jane\-php\/[a-z\-]*"\)\:\s\"\^[0-9\.]*\"/\1\: \"\*\@dev\"/' composer.json
+
+    cd -
+}
+export -f use_path_repository_in_composer
+
+install_component () {
+    printf "🌱 $1 👇\n"
+    cd $1 ; composer update --no-progress --no-suggest --ansi --prefer-stable $2 ; cd -
+}
+export -f install_component
+
+for i in $(find src -maxdepth 2 -type f -name composer.json -printf '%h\n' | sort); do
+    use_path_repository_in_composer "$i"
```

Comment:
I don't understand why you are using a path here. It's almost the same as testing the whole library

Reply:
The idea is to test the components individually to avoid collisions in the tests (I've already had problems with identical namespaces for example) and also to keep the components as they are in the PR when we test them (so that if we test OpenApi3, we have the OpenApiCommon version of the PR for example).

Reply:
> The idea is to test the components individually to avoid collisions in the tests

Since you also test jane globally, this issue (which is not an issue BTW, all name should be unique) is not solved anyway.

I understand what you are trying to achieve, but I don't think this is the way to go. Because here, you don't test more things than with the global test suite. The only thing you test is the local composer.json (and it's not 100% accurate since you can get transitive deps)